addons/mail/static/src/discuss/core/web/thread_actions.js

```diff
@@ -2,6 +2,22 @@ import { registerThreadAction } from "@mail/core/common/thread_actions";
 
 import { _t } from "@web/core/l10n/translation";
 
+export const joinChannelAction = {
+    condition: ({ owner, thread }) =>
+        thread &&
+        !["chat", "group"].includes(thread.channel_type) &&
+        !thread.self_member_id &&
+        !owner.isDiscussSidebarChannelActions,
+    open: ({ store, thread }) =>
+        store.env.services.orm.call("discuss.channel", "add_members", [[thread.id]], {
+            partner_ids: [store.self.id],
+        }),
+    icon: "fa fa-fw fa-sign-in",
+    iconLarge: "fa fa-fw fa-lg fa-sign-in",
+    name: _t("Join Channel"),
+    sequence: 1,
```

Comment:
Would add `sequenceGroup: 5`, so it's in the same group as `expand-discuss` which is at the top in chat window.

Channel sub-threads can be open in chat window without being member, so this should also be over there. I think it makes more sense to keep `Open in Discuss` as the 1st action and "Join Channel" just below, so I would put `sequence: 20` so this is above `expand-discuss` at `10` (note: sequence is local to a sequenceGroup, so could have sequence: 999999 and sequenceGroup: 10, this is just the last item in the 1st group).

The sequence numbers are quite low, so I'll probably bump all of them to have more room for choice.